ui/src/components/WebRTCVideo.tsx

```diff
@@ -144,27 +169,39 @@ export default function WebRTCVideo() {
     return () => {
       abortController.abort();
     };
-  }, [isPointerLockPossible, videoElm]);
+  }, [isPointerLockPossible]);
 
   const requestFullscreen = useCallback(async () => {
-    videoElm.current?.requestFullscreen({
+     if (!isFullscreenEnabled || !videoElm.current) return;
+
+    // per https://wicg.github.io/keyboard-lock/#system-key-press-handler
+    // If keyboard lock is activated after fullscreen is already in effect, then the user my 
+    // see multiple messages about how to exit fullscreen. For this reason, we recommend that
+    // developers call lock() before they enter fullscreen:
+    await requestKeyboardLock();
+    await requestPointerLock();
+
+    await videoElm.current.requestFullscreen({
```

Comment:
... and we have to wait for the transition to full-screen before moving on to other things...